<font color=blue>TITLE</font>

Mobile App for Gambling Addiction

<font color=blue>DESCRIPTION</font>

This project is to use combinations calculations to help gambling addicts see clearly how low the probabilities are of winning at the lottery.

In [4]:
import random

In [3]:
#functions for combination
def factorial(n):
    final_product = 1
    for i in range(n, 0, -1):
        final_product *= i
    return final_product
def combinations(n,k):
    return(factorial(n)/(factorial(k)*factorial(n-k)))
           

In [3]:
#function for probability of having a winning ticket

def one_ticket_probability(n,k,tnumbers):
    c_49_6 = combinations(49,6)
    print("If your ticket numbers were:", tnumbers, '\n')
    print("The probability of you winning is:", '{:.2E}'.format(1/c_49_6), '\n')
    print('''This means you have a 1 in {:,} chance of winning'''.format(c_49_6))

In [4]:
#randomly generate 6 ticket nos. w/o replacement, calc winning probability
user_number_list = random.sample(range(1,50), 6)
one_ticket_probability(49,6, user_number_list)

If your ticket numbers were: [14, 17, 26, 1, 31, 45] 

The probability of you winning is: 7.15E-08 

This means you have a 1 in 13,983,816.0 chance of winning


In [ ]:
import pandas as pd
data_649 = pd.read_csv('649.csv')
print(data_649.shape)
print(data_649.head(3))
data_649.tail(3)

In [ ]:
data_649.columns

In [7]:
#practice grabbing just one row and combining the
#numbers into a set

row = set(data_649.loc[1, 'NUMBER DRAWN 1': 
                       'NUMBER DRAWN 6'])
row

{8, 33, 36, 37, 39, 41}

In [10]:
#loop the above, creating a new series

values = []
def extract_numbers():
    for row in range(0,len(data_649)-1):
        value = set(data_649.loc[row, 'NUMBER DRAWN 1': 
                             'NUMBER DRAWN 6' ])
        values.append(value)
    return values
    
#check that this worked
extract_numbers()
values[0:3] 

[{3, 11, 12, 14, 41, 43}, {8, 33, 36, 37, 39, 41}, {1, 6, 23, 24, 27, 39}]

In [26]:
def set_in_list(my_list,iters):
    true_count = 0
    loops = 0
    my_set = set(random.sample(range(0,49),6))
    for i in range(iters):   
        for rows in my_list:
            if my_set == rows:
                true_count +=1
        loops +=1
    print('''How many lottery numbers were checked
    to see if they were drawn previously in the 
    lottery: ''', loops)
    print('''How many of those numbers had been drawn
    previously in the lottery: ''', true_count)
    return

In [27]:

set_in_list(values,1000)
    

How many lottery numbers were checked
    to see if they were drawn previously in the 
    lottery:  1000
How many of those numbers had been drawn
    previously in the lottery:  0


<font color = blue>EXPLANATION</font>

The app has randomly chosen the indicated number of lottery numbers and checked to see how many of those numbers had been drawn previously in the lottery.  As you can see, 1000 lottery numbers were randomly chosen and none of them had been drawn previously in the lottery.

In [13]:
def multi_ticket_probability(n_tickets):
    poss_combs = combinations(49,6)
    prob = n_tickets/poss_combs
    print('''The chance of winning with {:,} ticket(s)'''.format(n_tickets),
          'is 1 in {:,}'.format(int(poss_combs/n_tickets)))
    return

In [14]:
multi_ticket_probability(1)
multi_ticket_probability(10)
multi_ticket_probability(100)
multi_ticket_probability(1000)
multi_ticket_probability(1000000)
multi_ticket_probability(6991908)
multi_ticket_probability(13983816)

The chance of winning with 1 ticket(s) is 1 in 13,983,816
The chance of winning with 10 ticket(s) is 1 in 1,398,381
The chance of winning with 100 ticket(s) is 1 in 139,838
The chance of winning with 1,000 ticket(s) is 1 in 13,983
The chance of winning with 1,000,000 ticket(s) is 1 in 13
The chance of winning with 6,991,908 ticket(s) is 1 in 2
The chance of winning with 13,983,816 ticket(s) is 1 in 1


<font color = blue>EXPLANATION</font>

The app is showing your chance of winning based on the number of tickets you buy. Say you buy 10 tickets and each ticket co
sts 1 dollar. You will spend $10 and have an approximately 1 in a million chance of winning.

If you spend $1000 for 1000 tickets you still have only an approximately 1 in a 14,000 chance of winning.  

<font color = blue>NOTE:</font> 

The next section determines the probability of matching only some of the numbers on a ticket. Lotteries allow you to win some money this way.  To calculate this probability requires multiplying the number of successful matching combinations by the number of possible combinations of the remaining numbers.

In [7]:
def probability_less_6(n,k):
    #n- numbers drawn, k- number of required matches
    combs = combinations(n,k) # no of succesful combinations
    combs1 = combinations(49 - k, 6 - k) # no of combinations for remaining numbers
    success = combs * combs1
    prob = success/combinations(49,6)
    return print('Chance of matching',k,'numbers is 1 in', int(1/prob))

In [12]:
probability_less_6(6,2)
probability_less_6(6,3)
probability_less_6(6,4)
probability_less_6(6,5)

Chance of matching 2 numbers is 1 in 5
Chance of matching 3 numbers is 1 in 46
Chance of matching 4 numbers is 1 in 941
Chance of matching 5 numbers is 1 in 52969


<font color = blue>EXPLANATION</font>
This describes your chance of having a winning ticket based on the number of numbers you need to match. For example, the chance of you matching 4 lottery numbers with your ticket is 1 in 941. Another way to think of this is if 941 people buy a ticket only 1 of them will match 4 of the lottery numbers. 